### PR TITLE
Feat/waitlist api endpoint

### PR DESCRIPTION
**Pull request by gurumarv**
**How This PR Should Be Tested**
This PR should be tested by verifying the following:

- The endpoint correctly handles valid and invalid user data.
- Data is properly stored in the database upon successful signup.
- Confirmation emails are sent to users who successfully sign up.
- The API implements rate limiting and returns the appropriate status codes when the limit is exceeded.
- Error handling is appropriate and returns correct status codes and messages.

**How to Test**

**1. Valid Submission Test**

-   Send a POST request to /api/v1/waitlist with a valid JSON body containing email and fullName.
-   Expected response: HTTP 201 with message "You are all signed up!".
-   Verify that the user data is stored in the database.
-   Check that a confirmation email is sent to the provided email address.

**2. Invalid Data Test**

-   Send POST requests with various invalid data (e.g., missing fields, invalid email format) to /api/v1/waitlist.
-   Expected response: HTTP 400 with an appropriate error message (e.g., "Invalid email format").

**3. Rate Limiting Test**

-   Simulate multiple requests in a short period to exceed the rate limit.
-   Expected response: HTTP 429 with a message indicating that the rate limit has been exceeded.

**4. Error Handling Test**

-   Test the endpoint's response to various failure scenarios, such as database errors or invalid request formats.
-   Ensure appropriate status codes and error messages are returned.

**Checklist of What Was Implemented**
   

- [x] Endpoint Creation: Developed the /api/v1/waitlist endpoint to accept POST requests.
- [x]    Validation and Sanitization: Implemented validation rules for email format, required fields, and non-empty names.
- [x]    Data Storage: Securely stored user information in the database upon successful validation.
- [x]    Confirmation Email: Integrated the email service to send confirmation emails to users upon successful signup.
- [x]    Rate Limiting: Added rate limiting to prevent abuse of the signup endpoint.
- [x]    Error Handling: Implemented comprehensive error handling for invalid data, rate limiting, and internal errors.

**Reference Issue**
[FEAT: Waitlist API Bckend](https://github.com/hngprojects/hng_boilerplate_java_web/issues/2)

### CODE REVIEW

File: src/main/resources/application.properties
Comment (Am0du) [inline src/main/resources/application.properties:1]: Don't commit applications.properties

File: src/main/resources/db/migration/V1__init_database.sql [hunk 1]
```diff
@@ -39,4 +42,11 @@ CREATE TABLE user_organisation (
     PRIMARY KEY (user_id, organisation_id),
     FOREIGN KEY (user_id) REFERENCES users(id),
     FOREIGN KEY (organisation_id) REFERENCES organisations(id)
-);
\ No newline at end of file
+);
```
Comment (Am0du) [inline src/main/resources/db/migration/V1__init_database.sql:45]: Can you create another version for your migration v8 is available

**Review by Am0du — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 2]
```diff
@@ -157,20 +152,13 @@
...
-
-	<repositories>
-		<repository>
-			<id>central</id>
-			<url>https://repo.maven.apache.org/maven2</url>
-		</repository>
-	</repositories>	
-</project>
\ No newline at end of file
+</project>
```
Comment (lurldgbodex) [inline pom.xml:164]: I can see that you removed the springboot plugin. hope this is not going to cause a break?

**Review by lurldgbodex — APPROVED** [review]

**Comment by Am0du** [comment]
LGTM

**Comment by Am0du** [comment]
fw

**Review by Am0du — APPROVED** [review]

**Pull request merged**